src/webview/mod.rs

```diff
@@ -160,6 +160,26 @@ pub struct WebViewAttributes {
   /// allow to navigate and false is not.
   pub navigation_handler: Option<Box<dyn Fn(String) -> bool>>,
 
+  /// Set a download started handler to manage incoming downloads.
+  ///
+  /// The closure takes two parameters - the first is a `String` representing the url being downloaded from and and the
+  /// second is a mutable `PathBuf` reference that (possibly) represents where the file will be downloaded to. The latter
+  /// parameter can be used to set the download location by assigning a new path to it - the assigned path _must_ be
+  /// absolute. The closure returns a `bool` to allow or deny the download.
+  pub download_started_handler: Option<Box<dyn FnMut(String, &mut PathBuf) -> bool>>,
+
+  /// Sets a download completion handler to manage downloads that have finished.
+  ///
+  /// The closure is fired when the download completes, whether it was successful.
+  /// The closure takes a `String` representing the URL of the original download request, a `String` representing the
+  /// filesystem path the file was downloaded to (if successful), and a `bool` indiciating if the download succeeded.
+  ///
+  /// ## Platform-specific:
+  ///
+  /// - **macOS**: The second parameter indicating the path the file was saved to is always empty, due to API
+  /// limitations.
+  pub download_completed_handler: Option<Rc<dyn Fn(String, String, bool) + 'static>>,
```

Comment:
could we change this also to provide `PathBuf`
```suggestion
  pub download_completed_handler: Option<Rc<dyn Fn(String, PathBuf, bool) + 'static>>,
```